setup.py

```diff
@@ -1,10 +1,23 @@
 from __future__ import with_statement
+import os
 import sys
 
 from setuptools import setup, find_packages
 from setuptools.command.test import test as TestCommand
 
-from pyoos import __version__
+def extract_version(module='pyoos'):
```

Comment:
This pattern avoid importing the module before installing it.  It does add some complexity, but it helps some packaging tools.  For example: if this is not there we need to install all the dependencies at build time, which is not always desirable.

However, that is not essential and not the main point of this PR.  I can remove that if you wish.